### ISSUE TITLE
Add compareTo operator for XInt/YInt

### ISSUE THREAD

**Issue by dimsuz** [cashapp/contour#63]
There are certain scenarios (like the one described in #59) in which it is handy to do something like this:

```
if (view1.bottom() < view2.bottom()) view1.bottom() else otherView.top()
```

Currently `XInt`/`YInt` lack `operator fun compareTo` which prevents the above from compiling.
The workaround could be to use `minOf(view1.bottom() - view2.bottom(), 0.ydip) == 0.ydip` but that's a bit ugly.

Can those be added (I can forge a PR) or do you have something specific in mind why this is not a good idea?

**Comment by carranca** [comment]
That feels like an omission - seems like it'd be a good addition (for XInt/YInt/XFloat/YFloat)

**Comment by carranca** [comment]
@dimsuz a PR would definitely be welcome, otherwise I or someone else might try to get to it over the next week or two?

**Comment by dimsuz** [comment]
Yep, I'll post one unless someone beats me to it.
I guess it's  nothing complicated along the lines of

```kotlin
operator fun compareTo(other: XInt) {
  return this.value.compareTo(other.value)
}
```
etc

**Comment by oldergod** [comment]
I will do it right now

**Comment by oldergod** [comment]
Released in 0.1.7
